lib/actions/local/configure.coffee

```diff
@@ -95,8 +66,48 @@ inquirerOptions = (data) -> [
 	}
 ]
 
+# Since these paths are used by `resin-image-fs`, we can safely
+# assume they are always UNIX style.
+joinPath = (path) -> '/' + path.join('/')
+
+loadReconfix = (target) ->
+	{ Reconfix } = require('reconfix-preview')
+	imagefs = require('resin-image-fs')
+
+	reconfix = new Reconfix({
+		read: (partition, path) ->
+			imagefs.readFile
+				image: target
+				partition: partition
+				path: joinPath(path)
+			.catch (err) ->
+				if err.code == 'NOENT'
+					return Buffer.alloc(0)
+				else
+					throw err
+			.then (content) -> Buffer.from(content, 'utf-8')
+			.catch (err) -> {}
+		write: (partition, path, data) ->
+			imagefs.writeFile
+				image: target,
+				partition: partition,
+				path: joinPath(path)
+			, data.toString('utf-8')
+	})
+	schema = require('../../../extras/schemas/resin-os.json')
+	reconfix.loadSchema(schema)
+	Promise.resolve(reconfix)
+
+readConfiguration = (target) ->
```

Comment:
These 2 functions are only used in one place as far as I can see, so maybe worth inlining them?